packages/features/ee/organizations/lib/onboardingStore.ts

```diff
@@ -100,18 +123,102 @@ export const useOnboardingStore = create<OnboardingStoreState>()(
       // Reset action
       reset: (state) => {
         if (state) {
-          set(state);
+          const currentId = get().onboardingId;
+          const newId = state.onboardingId;
+
+          // If onboardingId is changing, handle localStorage migration
+          if (typeof window !== "undefined" && currentId !== newId) {
+            // Remove old key if it exists
+            if (currentId) {
+              localStorage.removeItem(getStorageKey(currentId));
+            } else {
+              localStorage.removeItem("org-creation-onboarding");
+            }
+
+            // Set new state which will be persisted with new key
+            set(state);
+
+            // Immediately save to new key
+            if (newId) {
+              const newKey = getStorageKey(newId);
+              localStorage.setItem(newKey, JSON.stringify({ state, version: 0 }));
+            }
+          } else {
+            set(state);
+          }
         } else {
           // Clear the localStorage entry for this store
           if (typeof window !== "undefined") {
-            localStorage.removeItem("org-creation-onboarding");
+            const currentId = get().onboardingId;
+            if (currentId) {
+              localStorage.removeItem(getStorageKey(currentId));
+            } else {
+              localStorage.removeItem("org-creation-onboarding");
```

Comment:
We could just do this as _getStorageKey_ handles both keys
```suggestion
           localStorage.removeItem(getStorageKey(currentId))
```